scripts/prepare-npm-package.js

```diff
@@ -1,105 +1,75 @@
-  }
-
-  // Fix for @grafana/i18n so eslint-plugin can be imported by consumers
-  if (pkgJson.content.name === '@grafana/i18n') {
-    exports['./eslint-plugin'] = {
-      import: './dist/eslint/index.cjs',
-      require: './dist/eslint/index.cjs',
-    };
-  }
-
-  pkgJson.update({
-    main: cjsIndex,
-    types: typesIndex,
-    module: esmIndex,
-    exports,
-  });
-
-  await pkgJson.save();
-
-  // If an alias package name is provided we add an exports entry for the alias
-  // then generate an additional "nested" package.json for typescript resolution that
-  // doesn't use the exports property in package.json.
-  if (process.env.ALIAS_PACKAGE_NAME) {
-    const aliasNames = process.env.ALIAS_PACKAGE_NAME.split(',');
+  // Aliasable exports point to a separate entry point. The following should not be aliased:
+  // - ./package.json
+  // - .
+  // - ./index
+  // - Wildcard exports
+  // - exports with a single key '@grafana-app/source'
+  const aliasableExports = Object.entries(pkgJsonExports).reduce((acc, [key, val]) => {
+    if (key === './package.json' || key === '.' || key === './index' || key.endsWith('*')) {
+      return acc;
+    }
 
-    const additionalExports = aliasNames.reduce((acc, alias) => {
-      acc[`./${alias}`] = {
-        import: {
-          types: typesIndex.replace('index', alias),
-          default: esmIndex.replace('index', alias),
-        },
-        require: {
-          types: typesIndex.replace('index', alias),
-          default: cjsIndex.replace('index', alias),
-        },
-      };
+    if (
+      val !== null &&
+      typeof val === 'object' &&
+      Object.keys(val).length === 1 &&
+      Object.keys(val)[0] === '@grafana-app/source'
+    ) {
       return acc;
-    }, {});
+    }
 
-    pkgJson.update({
-      exports: {
-        ...pkgJson.content.exports,
-        ...additionalExports,
-      },
-      files: [...pkgJson.content.files, ...aliasNames],
-    });
-    await pkgJson.save();
+    acc[key] = val;
+    return acc;
+  }, {});
 
-    for await (const aliasName of aliasNames) {
-      await createAliasPackageJsonFiles(pkgJson.content, aliasName);
-    }
+  // For backwards compatibility generate additional "nested" package.json for
+  // typescript resolutions that don't support exports. e.g. node10.
+  for (const [aliasName, pkgJsonExport] of Object.entries(aliasableExports)) {
+    await createAliasPackageJsonFiles(pkgJson.content, aliasName, pkgJsonExport);
   }
 } catch (e) {
   console.error(e);
   process.exit(1);
 }
 
-async function createAliasPackageJsonFiles(packageJsonContent, aliasName) {
-  const pkgName = `${packageJsonContent.name}/${aliasName}`;
+async function createAliasPackageJsonFiles(packageJsonContent, aliasName, pkgJsonExport) {
+  const pkgName = join(packageJsonContent.name, aliasName);
   try {
     console.log(`📦 Writing alias package.json for ${pkgName}.`);
-    const pkgJsonPath = `${cwd}/${aliasName}`;
+    const pkgJsonPath = join(cwd, aliasName);
```

Comment:
Does this need to also include the package.module path? I don't have a full understanding of this but I think it should so it ends up in the package? I'm seeing package.json files created in the wrong place otherwise:

```
const pkgJsonPath = join(cwd, dirname(packageJsonContent.module), aliasName);
```
(or similar)